File: 1.two-sum.go
/*
 * Runtime: 4 ms
 * Memory Usage: 3.4 MB
 */
func twoSum(nums []int, target int) []int {
    m := make(map[int]int, len(nums))
    for i, num := range nums {
        if v, found := m[target - num]; found {
            return []int{v,i}
        }
        m[num] = i
    }    
    return nil
}

File: 1.two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 *
 * Runtime: 56 ms
 * Memory Usage: 34.3 MB
 */
var twoSum = function(nums, target) {
    const hashmap = {}
    for (let i = 0, l = nums.length; i < l; i++) {
        const complement = target - nums[i];
        if (hashmap[complement] !== undefined) {
            return [hashmap[complement], i]
        }
        hashmap[nums[i]] = i
    }
    
    return []
}
